pkg/services/ngalert/api/forked_am.go

```diff
@@ -0,0 +1,129 @@
+package api
+
+import (
+	"fmt"
+
+	apimodels "github.com/grafana/alerting-api/pkg/api"
+	"github.com/grafana/grafana/pkg/api/response"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/datasources"
+)
+
+type ForkedAMSvc struct {
+	AMSvc, GrafanaSvc AlertmanagerApiService
+	DatasourceCache   datasources.CacheService
+}
+
+func NewForkedAM(datasourceCache datasources.CacheService, proxy, grafana AlertmanagerApiService) *ForkedAMSvc {
+	return &ForkedAMSvc{
+		AMSvc:           proxy,
+		GrafanaSvc:      grafana,
+		DatasourceCache: datasourceCache,
+	}
+}
+
+func (am *ForkedAMSvc) getService(ctx *models.ReqContext, datasourceCache datasources.CacheService) (AlertmanagerApiService, error) {
+	t, err := backendType(ctx, datasourceCache)
```

Comment:
There's no reason to require a parameter that's already on the receiver (`am`).
```suggestion
func (am *ForkedAMSvc) getService(ctx *models.ReqContext) (AlertmanagerApiService, error) {
	t, err := backendType(ctx, am.DatasourceCache)
```